### PR TITLE
[onert] Add unit tests for training Reshape op

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds some tests to validate training of Reshape op.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+  _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+  _context->addTrainCase(
+    uniformTCD<float>({{{4, 0,  -5, 1, 0,  4, -1, 1, -1, -3, 3,  -2, -4,
+                         1, -2, 2,  4, -4, 2, 2,  0, 4,  -1, -2, 4}}}, // input dataset
+                      {{{47, -4, -25, 9, 10, 10, -13, 11, -14, -26, -12, 26, 20, 40, 1, 3, 11,
+                         4}}},    // expected dataset
+                      {226.5260f} // last losses
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc:52]: The result of executing the same model with the same data on tensorflow:
```
Epoch 1/5
1/1 [==============================] - 0s 163ms/step - loss: 403.3333 - categorical_accuracy: 1.0000
Epoch 2/5
1/1 [==============================] - 0s 2ms/step - loss: 324.0978 - categorical_accuracy: 0.0000e+00 
Epoch 3/5 
1/1 [==============================] - 0s 2ms/step - loss: 267.7882 - categorical_accuracy: 0.0000e+00 
Epoch 4/5 
1/1 [==============================] - 0s 2ms/step - loss: 226.5260 - categorical_accuracy: 0.0000e+00 
Epoch 5/5 
1/1 [==============================] - 0s 2ms/step - loss: 195.3313 - categorical_accuracy: 0.0000e+00
```

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  cgen.addOperatorConv2D({{input, weight, bias}, {conv_output}}, circle::Padding_VALID, 1, 1,
+                         circle::ActivationFunctionType_NONE, 1, 1);
+  cgen.addOperatorReshape({{conv_output, shape}, {output}}, &new_shape);
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc:44]: If #13199 is merged first,
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  // Invalid shape: The number of output elements should be equal to input
+  int output = cgen.addTensor({{2, 3}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorReshape({{input, shape}, {output}});
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc:78]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  int shape = cgen.addTensor({{2}, circle::TensorType::TensorType_INT32, shape_buf});
+  int output = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorReshape({{input, shape}, {output}});
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Reshape.test.cc:101]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

**Comment by ragmani** [comment]
Added `cgen.markAllOpsAsTrainable();`

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by zetwhite — APPROVED** [review]
LGTM :+1:

**Pull request merged**